nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/test/java/org/apache/nifi/processors/aws/sns/ITPutSNS.java

```diff
@@ -36,43 +45,75 @@
  */
 public class ITPutSNS {
 
-    private static final String CREDENTIALS_FILE = System.getProperty("user.home") + "/aws-credentials.properties";
-    private final String TOPIC_ARN = "Add SNS ARN here";
+    private static final DockerImageName localstackImage = DockerImageName.parse("localstack/localstack:latest");
+
+    private static final LocalStackContainer localstack = new LocalStackContainer(localstackImage)
+            .withServices(LocalStackContainer.Service.SNS);
+
+    private static final String CREDENTIALS_FILE = "src/test/resources/mock-aws-credentials.properties";
+    private static String topicARN;
+    private static SnsClient client;
 
     @BeforeAll
-    public static void assumeCredentialsFileExists() {
-        Assumptions.assumeTrue(new File(CREDENTIALS_FILE).exists());
+    public static void setup() throws InterruptedException {
+        System.setProperty("software.amazon.awssdk.http.service.impl", "software.amazon.awssdk.http.urlconnection.UrlConnectionSdkHttpService");
+        localstack.start();
+
+        client = SnsClient.builder()
+                .endpointOverride(localstack.getEndpoint())
+                .credentialsProvider(
+                        StaticCredentialsProvider.create(
+                                AwsBasicCredentials.create(localstack.getAccessKey(), localstack.getSecretKey())
+                        )
+                )
+                .region(Region.of(localstack.getRegion()))
+                .build();
+
+        final CreateTopicResponse response = client.createTopic(CreateTopicRequest.builder()
+                .name("SnsSystemTest")
+                .build());
+        assertTrue(response.sdkHttpResponse().isSuccessful());
+        topicARN = response.topicArn();
+    }
+
+    @AfterAll
+    public static void shutdown() {
+        client.close();
+        localstack.stop();
     }
 
     @Test
     public void testPublish() throws IOException {
-        final TestRunner runner = TestRunners.newTestRunner(new PutSNS());
-        AuthUtils.enableCredentialsFile(runner, CREDENTIALS_FILE);
-        runner.setProperty(PutSNS.ARN, TOPIC_ARN);
+        final TestRunner runner = initRunner(PutSNS.class);
+        AuthUtils.enableAccessKey(runner, localstack.getAccessKey(), localstack.getSecretKey());
```

Comment:
@sujkm I cannot say I know all the details but I am not sure why this test and the other test were modified to always have `ENDPOINT_OVERRIDE`? Shouldn't there be two other tests with this setting and this test and the other test kept as they were?

Reply:
@dan-s1 Currently the tests were being ignored if there was no credentials file. Based on NIFI-12144 i thought it would be preferable to use test containers and TestPutSNS would still test without the endpoint override.

Reply:
Correct me if I am wrong, the way it is set up now with the latest changes, TestPutSNS only tests with the endpoint override.

Reply:
The current changes are only applied to the integration tests in ITPutSNS. TestPutSNS does not use the endpoint override.

Reply:
@sujkm Is it possible to have a test in TestPutSNS to test the endpoint override?

Reply:
I'm not sure how to test this since all the current examples of testing endpoint override are done in the integration tests. Should I leave the integration test as it was?

Reply:
I did not realize that. Keep your changes if this is the only way to test the override.